### PR TITLE
S03 us13 inscription

### PR DESCRIPTION
**Pull request by jaldabaoth-code**

### CODE REVIEW

**Review by RedPandore — DISMISSED** [review]

File: migrations/Version20210621074133.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE user ADD user_name VARCHAR(255) NOT NULL');
+    }
+
+    public function down(Schema $schema): void
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE user DROP user_name');
+    }
+}
```
Comment (LucyMerlier) [inline migrations/Version20210621074133.php]: doesn't this migration do exactly the same thing as the previous one?..

File: migrations/Version20210621090216.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        $this->addSql('CREATE TABLE subscription (id INT AUTO_INCREMENT NOT NULL, subscription_at DATETIME NOT NULL, curriculum VARCHAR(255) NOT NULL, job_description LONGTEXT DEFAULT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('CREATE TABLE user (id INT AUTO_INCREMENT NOT NULL, email VARCHAR(180) NOT NULL, roles JSON NOT NULL, password VARCHAR(255) NOT NULL, created_at DATETIME NOT NULL, username VARCHAR(255) NOT NULL, UNIQUE INDEX UNIQ_8D93D649E7927C74 (email), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE registered_user ADD CONSTRAINT FK_8B903F56A76ED395 FOREIGN KEY (user_id) REFERENCES user (id)');
+        $this->addSql('ALTER TABLE registered_user ADD CONSTRAINT FK_8B903F569A1887DC FOREIGN KEY (subscription_id) REFERENCES subscription (id)');
+    }
+
```
Comment (LucyMerlier) [inline migrations/Version20210621090216.php]: didn't you already have at least some of those tables?

File: migrations/Version20210621095727.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        $this->addSql('CREATE TABLE subscription (id INT AUTO_INCREMENT NOT NULL, subscription_at DATETIME NOT NULL, curriculum VARCHAR(255) NOT NULL, job_description LONGTEXT DEFAULT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('CREATE TABLE user (id INT AUTO_INCREMENT NOT NULL, email VARCHAR(180) NOT NULL, roles JSON NOT NULL, password VARCHAR(255) NOT NULL, created_at DATETIME NOT NULL, username VARCHAR(255) NOT NULL, UNIQUE INDEX UNIQ_8D93D649E7927C74 (email), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE registered_user ADD CONSTRAINT FK_8B903F56A76ED395 FOREIGN KEY (user_id) REFERENCES user (id)');
+        $this->addSql('ALTER TABLE registered_user ADD CONSTRAINT FK_8B903F569A1887DC FOREIGN KEY (subscription_id) REFERENCES subscription (id)');
+    }
```
Comment (LucyMerlier) [inline migrations/Version20210621095727.php]: this does the same as above

File: migrations/Version20210621101252.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        $this->addSql('CREATE TABLE subscription (id INT AUTO_INCREMENT NOT NULL, subscription_at DATETIME NOT NULL, curriculum VARCHAR(255) NOT NULL, job_description LONGTEXT DEFAULT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('CREATE TABLE user (id INT AUTO_INCREMENT NOT NULL, email VARCHAR(180) NOT NULL, roles JSON NOT NULL, password VARCHAR(255) NOT NULL, created_at DATETIME NOT NULL, username VARCHAR(255) NOT NULL, UNIQUE INDEX UNIQ_8D93D649E7927C74 (email), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE registered_user ADD CONSTRAINT FK_8B903F56A76ED395 FOREIGN KEY (user_id) REFERENCES user (id)');
+        $this->addSql('ALTER TABLE registered_user ADD CONSTRAINT FK_8B903F569A1887DC FOREIGN KEY (subscription_id) REFERENCES subscription (id)');
+    }
```
Comment (LucyMerlier) [inline migrations/Version20210621101252.php]: this does the same as above

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by RaphaelBS-WCS — APPROVED** [review]
Approved

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**